include/ncutility/detail/EnumerateDetail.h

```diff
@@ -0,0 +1,364 @@
+#pragma once
+
+// Implementation of c++23 std::enumerate based on: https://github.com/cor3ntin/rangesnext
+
+#include <ranges>
+
+namespace nc::algo::detail {
+template<class... V>
+consteval auto iter_cat()
+{
+    if constexpr ((std::ranges::random_access_range<V> && ...))
+    {
+        return std::random_access_iterator_tag{};
+    }
+    else if constexpr ((std::ranges::bidirectional_range<V> && ...))
+    {
+        return std::bidirectional_iterator_tag{};
+    }
+    else if constexpr ((std::ranges::forward_range<V> && ...))
+    {
+        return std::forward_iterator_tag{};
+    }
+    else if constexpr ((std::ranges::input_range<V> && ...))
+    {
+        return std::input_iterator_tag{};
+    }
+    else
+    {
+        return std::output_iterator_tag{};
+    }
+}
+
+template<class R>
+concept simple_view =
+    std::ranges::view<R> &&
+    std::ranges::range<const R> &&
+    std::same_as<std::ranges::iterator_t<R>, std::ranges::iterator_t<const R>> &&
+    std::same_as<std::ranges::sentinel_t<R>, std::ranges::sentinel_t<const R>>;
+
+template<std::ranges::input_range V>
+    requires std::ranges::view<V>
+class enumerate_view : public std::ranges::view_interface<enumerate_view<V>> {
```

Comment:
How are you determining curly spacing re: newlines? I think it should be on a new line for class declarations

Reply:
Moved.